src/components/chatRooms/ChatRoomForm.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import TextInput from '../common/TextInput';
+import {Component} from 'react/lib/ReactBaseClasses';
+import ChatRoomApi from '../../api/chatRoom';
+
+class ChatRoomForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      chatRoom: {
+        name: ''
+      }
+    };
+
+    this.save = this.save.bind(this);
+    this.change = this.change.bind(this);
+  }
+
+  save(event) {
+    event.preventDefault();
+    ChatRoomApi.create(this.state.chatRoom);
+    const chatRoom = this.state.chatRoom;
+    chatRoom.name = '';
```

Comment:
⚠️ mutate data in react is very [not-recommanded](https://medium.com/pro-react/a-brief-talk-about-immutability-and-react-s-helpers-70919ab8ae7c).

Reply:
So instead, I should do something like this right ?
```javascript
const chatRoom = {name: ''};
this.setState({chatRoom});
```
And though never change directly something referenced in the `this.state` object ?

Reply:
Yes!